tests/test_ci_insights.py

```diff
@@ -86,22 +86,26 @@ def test_flaky_detection_detects_new_tests(
 
     result, spans = pytester_with_spans(
         code="""
+        import random
+
         def test_foo():
             assert True
 
         def test_bar():
-            assert True
+            # Simulate a flaky test.
+            assert random.choice([True, False])
```

Comment:
Don't write tests with random, there's always a chance This is going to be really "flaky" and you'll always get True 😆 
Better use e.g.

```python
counter = 0
def test_bar():
    global counter
    if counter == 2:
        pytest.fail("Oopsy i'm flaky")
    counter += 1
```